sources/pg_replication/helpers.py

```diff
@@ -0,0 +1,757 @@
+from typing import (
+    Optional,
+    Dict,
+    Iterator,
+    Union,
+    List,
+    Sequence,
+    Any,
+)
+from dataclasses import dataclass, field
+
+import psycopg2
+from psycopg2.extensions import cursor
+from psycopg2.extras import (
+    LogicalReplicationConnection,
+    ReplicationCursor,
+    ReplicationMessage,
+    StopReplication,
+)
+from pypgoutput.decoders import (  # type: ignore[import-untyped]
+    Begin,
+    Commit,
+    Relation,
+    Insert,
+    Update,
+    Delete,
+    Truncate,
+    ColumnData,
+)
+
+import dlt
+
+from dlt.common import logger
+from dlt.common.typing import TDataItem
+from dlt.common.pendulum import pendulum
+from dlt.common.schema.typing import (
+    TTableSchema,
+    TTableSchemaColumns,
+    TAnySchemaColumns,
+    TColumnNames,
+    TWriteDisposition,
+)
+from dlt.common.schema.utils import get_columns_names_with_prop
+from dlt.common.data_writers.escape import escape_postgres_identifier
+from dlt.common.configuration.specs import BaseConfiguration, configspec
+from dlt.extract.typing import DataItemWithMeta, TTableHintTemplate
+from dlt.extract.resource import DltResource
+from dlt.sources.credentials import ConnectionStringCredentials
+
+try:
+    from ..sql_database import sql_table  # type: ignore[import-untyped]
+except Exception:
+    from sql_database import sql_table
+
+from .schema_types import _to_dlt_column_schema, _to_dlt_val
+
+
+@dlt.sources.config.with_config(sections=("sources", "pg_replication"))
```

Comment:
Yes, but it only returns resources on the first call (when it creates the slot) and only if `persist_snapshots` is `True`. So I'd say it's primarily a helper to set up replication, rather than a source.